x/evm/keeper/abci.go

```diff
@@ -26,6 +27,11 @@ import (
 // BeginBlock sets the sdk Context and EIP155 chain id to the Keeper.
 func (k *Keeper) BeginBlock(ctx sdk.Context, _ abci.RequestBeginBlock) {
 	k.WithChainID(ctx)
+
+	if utils.IsEthermintDevChain(ctx) {
```

Comment:
U can do it on dymension as well. No need to hook it with epochs.

Reply:
@mtsitrin , Omri asked to use epoch, you can talk with him.
- BeginBlock would support instant deployment for new metadata, easier to test. But cost a little bit performance since it checked every blocks.
- Epoch hook cost less performance but harder to setup test. since having to wait.

Reply:
I think we plan to hook it with the bank metadata proposal, so I wouldn't invest too much effort on it
@omritoptix wdyt

Reply:
@mtsitrin makes sense. should be straight forward and the cleanest.